src/relations/db.py

```diff
@@ -136,6 +138,60 @@ def __init__(self, charm: CharmBase, admin: bool = False):
     def _depart_flag(self, relation):
         return f"{self.relation_name}_{relation.id}_departing"
 
+    def _check_extensions(self, extensions: List) -> bool:
+        """Checks if requested extensions are enabled."""
+        for extension in extensions:
+            extension_name = extension.split(":")[0]
+            if not self.charm.backend.database.is_postgresql_plugin_enabled(extension_name):
+                return False
+        return True
+
+    def _block_on_extensions(self, relation, remote_app_databag: Dict) -> bool:
+        """Verifies that extensions are enabled or blocks the charm."""
+        if "extensions" in remote_app_databag:
+            self.update_databags(
+                relation,
+                {
+                    "extensions": remote_app_databag["extensions"],
+                },
+            )
+            extensions = remote_app_databag["extensions"].split(",")
+            if not self._check_extensions(extensions):
+                # Do not allow apps requesting extensions to be installed.
+                logger.error(
+                    f"ERROR - `extensions` ({', '.join(extensions)}) cannot be requested through relations"
+                    " - Please enable extensions through `juju config` and add the relation again."
+                )
+                self.charm.unit.status = BlockedStatus(EXTENSIONS_BLOCKING_MESSAGE)
+                return False
+        return True
```

Comment:
Optional: I think we could return `True` when the charm blocks, to make the return of the method be more aligned with the method name. I got confused about the returned value in L229.